text/trace/0205-messaging-semantic-conventions-context-propagation.md

```diff
@@ -0,0 +1,173 @@
+
+## Terminology
+
+For terms used in this document, refer to [OTEP 173](#0173-messaging-semantic-conventions.md#terminology).
+
+## Motivation
+
+The current [messaging semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/a1a8676a43dce6a4e447f65518aef8e98784306c/specification/trace/semantic_conventions/messaging.md)
+provide a list of [examples](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#examples),
+which illustrate how producer and consumer spans can be correlated by
+parent/child relationships or links. All the examples assume that context
+information for a given message is propagated from the producer to the consumer.
+
+However, this is not a trivial assumption, and it is not easily accommodated by
+existing established context propagation mechanisms. Those mechanism propagate
+context on a per-request basis, whereas the messaging semantic conventions
+assume that context is propagated on a per-message basis. This means, that
+although several requests might be involved in the processing of a single
+message (publishing a message, fetching a message, probable multiple times by
+multiple consumers), it is assumed that all components have access to the same
+per-message context information that allows correlating all the stages of
+processing a message.
+
+To achieve this desired outcome, a context needs to be attached to a message,
+and intermediaries must not alter the context attached to the message. _This
+requirement should be documented, as it is an important factor in deciding how
+to propagate context for message scenarios, and how to standardize context
+propagation for existing message protocols._
+
+The additions proposed in this document are not breaking or invalidating any of
+the existing semantic conventions for messaging, but rather make an implicit
+requirement explicit.
+
+### Example
+
+Many intermediaries (message brokers) offer REST APIs for publishing and
+fetching messages. A producer can publish a message by sending an HTTP request
+to the intermediary, a consumer can pull a message by sending an HTTP request
+to the intermediary:
+
+```
+  +----------+
+  | Producer |
+  +----------+
+       |
+       | HTTP POST (publishing a message)
+       v
++--------------+
+| Intermediary |
++--------------+
+       ^
+       | HTTP GET (fetching a message)
+       |
+  +----------+
+  | Consumer |
+  +----------+
+```
+
+In this scenario, it is not possible to correlate producer and consumer traces
+for a given message by using HTTP context propagation mechanisms and without
+special broker instrumentation. Both HTTP requests, the one that publishes and
+the one that pulls the message, originate from different traces. There is no
+way the context attached to the HTTP POST request publishing the message can be
+forwarded by the HTTP GET request fetching a message.
```

Comment:
I would need to be forwarded _to_ the consumer, but it would need to be forwarded _by_ the HTTP GET request, when only using HTTP context propagation (without messaging specific context propagation).